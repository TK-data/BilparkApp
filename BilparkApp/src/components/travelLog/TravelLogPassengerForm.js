import React, { Component } from 'react';
import t from 'tcomb-form-native';
import { connect } from 'react-redux';
import { StyleSheet, View } from 'react-native';
import { travelLogFormValue, travelLogFormType } from '../../actions/travelLog';

export function getType(value) {

  const Passengers = t.enums({
    0: '0',
    1: '1',
    2: '2',
    3: '3',
    4: '4',
    5: '5',
  });

  if (value.Passenger === '1') {
    return t.struct({
      Passenger: Passengers,
      Passenger1: t.String,
    });
  } else if (value.Passenger === '2') {
    return t.struct({
      Passenger: Passengers,
      Passenger1: t.String,
      Passenger2: t.String,
    });
  } else if (value.Passenger === '3') {
    return t.struct({
      Passenger: Passengers,
      Passenger1: t.String,
      Passenger2: t.String,
      Passenger3: t.String,
    });
  } else if (value.Passenger === '4') {
    return t.struct({
      Passenger: Passengers,
      Passenger1: t.String,
      Passenger2: t.String,
      Passenger3: t.String,
      Passenger4: t.String,
    });
  } else if (value.Passenger === '5') {
    return t.struct({
      Passenger: Passengers,
      Passenger1: t.String,
      Passenger2: t.String,
      Passenger3: t.String,
      Passenger4: t.String,
      Passenger5: t.String,
    });
  }
  return t.struct({
    Passenger: Passengers,
  });

}

class TravelLogPassengerForm extends Component {

  onChange(value) {
    this.props.formValue(value);
    this.props.formType(getType(value));
  }

  render() {

    const Form = t.form.Form;

    const formOptions = {
      fields: {
        Passenger: {
          label: 'Antall passasjerer:',
          nullOption: { value: '', text: 'Velg antall Passasjerer' },

        },
        Passenger1: {
          label: 'Passasjer 1:',
        },
        Passenger2: {
          label: 'Passasjer 2:',
        },
        Passenger3: {
          label: 'Passasjer 3:',
        },
        Passenger4: {
          label: 'Passasjer 4:',
        },
        Passenger5: {
          label: 'Passasjer 5:',
        },
        Comment: {
          label: 'Kommentar (Valgfritt)',
          multiline: true,
          numberOfLines: 3,
        },
      },
    };

    return (
      <View style={styles.container}>
        <Form
          ref={c => this.form = c}
          type={this.props.type}
          options={formOptions}
          value={this.props.value}
          onChange={value => this.onChange(value)}
        />
      </View>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    isLoading: state.auth.isLoading,
    hasErrored: state.auth.hasErrored,
    user: state.auth.user,
    mail: state.loginMail,
    formOptions: state.loginOptions,
    value: state.travelLog.formValue,
    type: state.travelLog.formType,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    formValue: value => dispatch(travelLogFormValue(value)),
    formType: type => dispatch(travelLogFormType(type)),
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(TravelLogPassengerForm);


const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    marginTop: 10,
  },
  input: {
    height: 37,
    width: 250,
  },
  text: {
    color: '#fff',
  },
  button: {
    alignSelf: 'center',
  },
});
