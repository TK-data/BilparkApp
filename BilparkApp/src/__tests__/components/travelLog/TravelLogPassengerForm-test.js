import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import TravelLogPassengerForm from '../../../components/travelLog/TravelLogPassengerForm';

const initialState = {
  travelLog: {
    addressFrom: '',
    addressTo: '',
    cargoValue: {
      Cargo: false,
      Comment: '',
    },
    datepickerDate: '15.5.2018',
    datepickerVisible: false,
    distance: '0km',
    formType: true,
    formValue: {
      Passenger: '0',
      Passenger1: '',
      Passenger2: '',
      Passenger3: '',
      Passenger4: '',
      Passenger5: '',
    },
    positionFrom: '0',
    positionTo: '0',
  },
  hasErrored: false,
  isLoading: true,
  auth: {
    hasErrored: false,
    isLoading: false,
    user: 'user',
  },
};

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Testing TravelLogPassengerForm', () => {
  it('Renders as expected', () => {
    const comp = shallow(
      <TravelLogPassengerForm />,
      { context: { store: mockStore(initialState) } },
    );
    expect(comp.dive()).toMatchSnapshot();
  });

  const wrapper = shallow(
    <TravelLogPassengerForm />,
    { context: { store: mockStore(initialState) } },
  );
  it('Can use the onChange function', () => {
    const value = {
      Passenger: '1',
      Passenger1: '',
      Passenger2: '',
      Passenger3: '',
      Passenger4: '',
      Passenger5: '',
    };
    wrapper.dive().instance().onChange(value);
  });
  it('Can be passenger from 1 to 5', () => {
    const value = {
      Passenger: '1',
      Passenger1: '',
      Passenger2: '',
      Passenger3: '',
      Passenger4: '',
      Passenger5: '',
    };
    for (let i = 1; i < 6; i += 1) {
      value.Passenger = i.toString();
      wrapper.dive().instance().onChange(value);
    }
  });
});
