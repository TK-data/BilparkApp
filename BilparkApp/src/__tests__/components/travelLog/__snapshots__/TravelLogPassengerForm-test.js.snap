// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing TravelLogPassengerForm Renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TravelLogPassengerForm
    formOptions={undefined}
    formType={[Function]}
    formValue={[Function]}
    hasErrored={false}
    isLoading={false}
    mail={undefined}
    type={true}
    user="user"
    value={
      Object {
        "Passenger": "0",
        "Passenger1": "",
        "Passenger2": "",
        "Passenger3": "",
        "Passenger4": "",
        "Passenger5": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Form
        i18n={
          Object {
            "add": "Add",
            "down": "↓",
            "optional": " (optional)",
            "remove": "✘",
            "required": "",
            "up": "↑",
          }
        }
        onChange={[Function]}
        options={
          Object {
            "fields": Object {
              "Comment": Object {
                "label": "Kommentar (Valgfritt)",
                "multiline": true,
                "numberOfLines": 3,
              },
              "Passenger": Object {
                "label": "Antall passasjerer:",
                "nullOption": Object {
                  "text": "Velg antall Passasjerer",
                  "value": "",
                },
              },
              "Passenger1": Object {
                "label": "Passasjer 1:",
              },
              "Passenger2": Object {
                "label": "Passasjer 2:",
              },
              "Passenger3": Object {
                "label": "Passasjer 3:",
              },
              "Passenger4": Object {
                "label": "Passasjer 4:",
              },
              "Passenger5": Object {
                "label": "Passasjer 5:",
              },
            },
          }
        }
        stylesheet={
          Object {
            "button": Object {
              "alignSelf": "stretch",
              "backgroundColor": "#48BBEC",
              "borderColor": "#48BBEC",
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 36,
              "justifyContent": "center",
              "marginBottom": 10,
            },
            "buttonText": Object {
              "alignSelf": "center",
              "color": "white",
              "fontSize": 18,
            },
            "checkbox": Object {
              "error": Object {
                "marginBottom": 4,
              },
              "normal": Object {
                "marginBottom": 4,
              },
            },
            "controlLabel": Object {
              "error": Object {
                "color": "#a94442",
                "fontSize": 17,
                "fontWeight": "500",
                "marginBottom": 7,
              },
              "normal": Object {
                "color": "#000000",
                "fontSize": 17,
                "fontWeight": "500",
                "marginBottom": 7,
              },
            },
            "dateTouchable": Object {
              "error": Object {},
              "normal": Object {},
            },
            "dateValue": Object {
              "error": Object {
                "color": "#a94442",
                "fontSize": 17,
                "marginBottom": 5,
                "padding": 7,
              },
              "normal": Object {
                "color": "#000000",
                "fontSize": 17,
                "marginBottom": 5,
                "padding": 7,
              },
            },
            "datepicker": Object {
              "error": Object {
                "marginBottom": 4,
              },
              "normal": Object {
                "marginBottom": 4,
              },
            },
            "errorBlock": Object {
              "color": "#a94442",
              "fontSize": 17,
              "marginBottom": 2,
            },
            "fieldset": Object {},
            "formGroup": Object {
              "error": Object {
                "marginBottom": 10,
              },
              "normal": Object {
                "marginBottom": 10,
              },
            },
            "helpBlock": Object {
              "error": Object {
                "color": "#999999",
                "fontSize": 17,
                "marginBottom": 2,
              },
              "normal": Object {
                "color": "#999999",
                "fontSize": 17,
                "marginBottom": 2,
              },
            },
            "pickerContainer": Object {
              "error": Object {
                "borderColor": "#a94442",
                "borderRadius": 4,
                "borderWidth": 1,
                "marginBottom": 4,
              },
              "normal": Object {
                "borderColor": "#cccccc",
                "borderRadius": 4,
                "borderWidth": 1,
                "marginBottom": 4,
              },
              "open": Object {},
            },
            "pickerTouchable": Object {
              "active": Object {
                "borderBottomWidth": 1,
                "borderColor": "#cccccc",
              },
              "error": Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 44,
              },
              "normal": Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 44,
              },
            },
            "pickerValue": Object {
              "error": Object {
                "fontSize": 17,
                "paddingLeft": 7,
              },
              "normal": Object {
                "fontSize": 17,
                "paddingLeft": 7,
              },
            },
            "select": Object {
              "error": Object {},
              "normal": Object {},
            },
            "textbox": Object {
              "error": Object {
                "borderColor": "#a94442",
                "borderRadius": 4,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 17,
                "height": 36,
                "marginBottom": 5,
                "paddingHorizontal": 7,
                "paddingVertical": 7,
              },
              "normal": Object {
                "borderColor": "#cccccc",
                "borderRadius": 4,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 17,
                "height": 36,
                "marginBottom": 5,
                "paddingHorizontal": 7,
                "paddingVertical": 7,
              },
              "notEditable": Object {
                "backgroundColor": "#eeeeee",
                "borderColor": "#cccccc",
                "borderRadius": 4,
                "borderWidth": 1,
                "color": "#777777",
                "fontSize": 17,
                "height": 36,
                "marginBottom": 5,
                "paddingHorizontal": 7,
                "paddingVertical": 7,
              },
            },
            "textboxView": Object {
              "error": Object {},
              "normal": Object {},
              "notEditable": Object {},
            },
          }
        }
        templates={
          Object {
            "checkbox": [Function],
            "datepicker": [Function],
            "list": [Function],
            "select": [Function],
            "struct": [Function],
            "textbox": [Function],
          }
        }
        type={true}
        value={
          Object {
            "Passenger": "0",
            "Passenger1": "",
            "Passenger2": "",
            "Passenger3": "",
            "Passenger4": "",
            "Passenger5": "",
          }
        }
      />,
      "style": Object {
        "alignItems": "center",
        "flex": 1,
        "marginTop": 10,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "i18n": Object {
          "add": "Add",
          "down": "↓",
          "optional": " (optional)",
          "remove": "✘",
          "required": "",
          "up": "↑",
        },
        "onChange": [Function],
        "options": Object {
          "fields": Object {
            "Comment": Object {
              "label": "Kommentar (Valgfritt)",
              "multiline": true,
              "numberOfLines": 3,
            },
            "Passenger": Object {
              "label": "Antall passasjerer:",
              "nullOption": Object {
                "text": "Velg antall Passasjerer",
                "value": "",
              },
            },
            "Passenger1": Object {
              "label": "Passasjer 1:",
            },
            "Passenger2": Object {
              "label": "Passasjer 2:",
            },
            "Passenger3": Object {
              "label": "Passasjer 3:",
            },
            "Passenger4": Object {
              "label": "Passasjer 4:",
            },
            "Passenger5": Object {
              "label": "Passasjer 5:",
            },
          },
        },
        "stylesheet": Object {
          "button": Object {
            "alignSelf": "stretch",
            "backgroundColor": "#48BBEC",
            "borderColor": "#48BBEC",
            "borderRadius": 8,
            "borderWidth": 1,
            "height": 36,
            "justifyContent": "center",
            "marginBottom": 10,
          },
          "buttonText": Object {
            "alignSelf": "center",
            "color": "white",
            "fontSize": 18,
          },
          "checkbox": Object {
            "error": Object {
              "marginBottom": 4,
            },
            "normal": Object {
              "marginBottom": 4,
            },
          },
          "controlLabel": Object {
            "error": Object {
              "color": "#a94442",
              "fontSize": 17,
              "fontWeight": "500",
              "marginBottom": 7,
            },
            "normal": Object {
              "color": "#000000",
              "fontSize": 17,
              "fontWeight": "500",
              "marginBottom": 7,
            },
          },
          "dateTouchable": Object {
            "error": Object {},
            "normal": Object {},
          },
          "dateValue": Object {
            "error": Object {
              "color": "#a94442",
              "fontSize": 17,
              "marginBottom": 5,
              "padding": 7,
            },
            "normal": Object {
              "color": "#000000",
              "fontSize": 17,
              "marginBottom": 5,
              "padding": 7,
            },
          },
          "datepicker": Object {
            "error": Object {
              "marginBottom": 4,
            },
            "normal": Object {
              "marginBottom": 4,
            },
          },
          "errorBlock": Object {
            "color": "#a94442",
            "fontSize": 17,
            "marginBottom": 2,
          },
          "fieldset": Object {},
          "formGroup": Object {
            "error": Object {
              "marginBottom": 10,
            },
            "normal": Object {
              "marginBottom": 10,
            },
          },
          "helpBlock": Object {
            "error": Object {
              "color": "#999999",
              "fontSize": 17,
              "marginBottom": 2,
            },
            "normal": Object {
              "color": "#999999",
              "fontSize": 17,
              "marginBottom": 2,
            },
          },
          "pickerContainer": Object {
            "error": Object {
              "borderColor": "#a94442",
              "borderRadius": 4,
              "borderWidth": 1,
              "marginBottom": 4,
            },
            "normal": Object {
              "borderColor": "#cccccc",
              "borderRadius": 4,
              "borderWidth": 1,
              "marginBottom": 4,
            },
            "open": Object {},
          },
          "pickerTouchable": Object {
            "active": Object {
              "borderBottomWidth": 1,
              "borderColor": "#cccccc",
            },
            "error": Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 44,
            },
            "normal": Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 44,
            },
          },
          "pickerValue": Object {
            "error": Object {
              "fontSize": 17,
              "paddingLeft": 7,
            },
            "normal": Object {
              "fontSize": 17,
              "paddingLeft": 7,
            },
          },
          "select": Object {
            "error": Object {},
            "normal": Object {},
          },
          "textbox": Object {
            "error": Object {
              "borderColor": "#a94442",
              "borderRadius": 4,
              "borderWidth": 1,
              "color": "#000000",
              "fontSize": 17,
              "height": 36,
              "marginBottom": 5,
              "paddingHorizontal": 7,
              "paddingVertical": 7,
            },
            "normal": Object {
              "borderColor": "#cccccc",
              "borderRadius": 4,
              "borderWidth": 1,
              "color": "#000000",
              "fontSize": 17,
              "height": 36,
              "marginBottom": 5,
              "paddingHorizontal": 7,
              "paddingVertical": 7,
            },
            "notEditable": Object {
              "backgroundColor": "#eeeeee",
              "borderColor": "#cccccc",
              "borderRadius": 4,
              "borderWidth": 1,
              "color": "#777777",
              "fontSize": 17,
              "height": 36,
              "marginBottom": 5,
              "paddingHorizontal": 7,
              "paddingVertical": 7,
            },
          },
          "textboxView": Object {
            "error": Object {},
            "normal": Object {},
            "notEditable": Object {},
          },
        },
        "templates": Object {
          "checkbox": [Function],
          "datepicker": [Function],
          "list": [Function],
          "select": [Function],
          "struct": [Function],
          "textbox": [Function],
        },
        "type": true,
        "value": Object {
          "Passenger": "0",
          "Passenger1": "",
          "Passenger2": "",
          "Passenger3": "",
          "Passenger4": "",
          "Passenger5": "",
        },
      },
      "ref": [Function],
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Form
          i18n={
            Object {
              "add": "Add",
              "down": "↓",
              "optional": " (optional)",
              "remove": "✘",
              "required": "",
              "up": "↑",
            }
          }
          onChange={[Function]}
          options={
            Object {
              "fields": Object {
                "Comment": Object {
                  "label": "Kommentar (Valgfritt)",
                  "multiline": true,
                  "numberOfLines": 3,
                },
                "Passenger": Object {
                  "label": "Antall passasjerer:",
                  "nullOption": Object {
                    "text": "Velg antall Passasjerer",
                    "value": "",
                  },
                },
                "Passenger1": Object {
                  "label": "Passasjer 1:",
                },
                "Passenger2": Object {
                  "label": "Passasjer 2:",
                },
                "Passenger3": Object {
                  "label": "Passasjer 3:",
                },
                "Passenger4": Object {
                  "label": "Passasjer 4:",
                },
                "Passenger5": Object {
                  "label": "Passasjer 5:",
                },
              },
            }
          }
          stylesheet={
            Object {
              "button": Object {
                "alignSelf": "stretch",
                "backgroundColor": "#48BBEC",
                "borderColor": "#48BBEC",
                "borderRadius": 8,
                "borderWidth": 1,
                "height": 36,
                "justifyContent": "center",
                "marginBottom": 10,
              },
              "buttonText": Object {
                "alignSelf": "center",
                "color": "white",
                "fontSize": 18,
              },
              "checkbox": Object {
                "error": Object {
                  "marginBottom": 4,
                },
                "normal": Object {
                  "marginBottom": 4,
                },
              },
              "controlLabel": Object {
                "error": Object {
                  "color": "#a94442",
                  "fontSize": 17,
                  "fontWeight": "500",
                  "marginBottom": 7,
                },
                "normal": Object {
                  "color": "#000000",
                  "fontSize": 17,
                  "fontWeight": "500",
                  "marginBottom": 7,
                },
              },
              "dateTouchable": Object {
                "error": Object {},
                "normal": Object {},
              },
              "dateValue": Object {
                "error": Object {
                  "color": "#a94442",
                  "fontSize": 17,
                  "marginBottom": 5,
                  "padding": 7,
                },
                "normal": Object {
                  "color": "#000000",
                  "fontSize": 17,
                  "marginBottom": 5,
                  "padding": 7,
                },
              },
              "datepicker": Object {
                "error": Object {
                  "marginBottom": 4,
                },
                "normal": Object {
                  "marginBottom": 4,
                },
              },
              "errorBlock": Object {
                "color": "#a94442",
                "fontSize": 17,
                "marginBottom": 2,
              },
              "fieldset": Object {},
              "formGroup": Object {
                "error": Object {
                  "marginBottom": 10,
                },
                "normal": Object {
                  "marginBottom": 10,
                },
              },
              "helpBlock": Object {
                "error": Object {
                  "color": "#999999",
                  "fontSize": 17,
                  "marginBottom": 2,
                },
                "normal": Object {
                  "color": "#999999",
                  "fontSize": 17,
                  "marginBottom": 2,
                },
              },
              "pickerContainer": Object {
                "error": Object {
                  "borderColor": "#a94442",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "marginBottom": 4,
                },
                "normal": Object {
                  "borderColor": "#cccccc",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "marginBottom": 4,
                },
                "open": Object {},
              },
              "pickerTouchable": Object {
                "active": Object {
                  "borderBottomWidth": 1,
                  "borderColor": "#cccccc",
                },
                "error": Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "height": 44,
                },
                "normal": Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "height": 44,
                },
              },
              "pickerValue": Object {
                "error": Object {
                  "fontSize": 17,
                  "paddingLeft": 7,
                },
                "normal": Object {
                  "fontSize": 17,
                  "paddingLeft": 7,
                },
              },
              "select": Object {
                "error": Object {},
                "normal": Object {},
              },
              "textbox": Object {
                "error": Object {
                  "borderColor": "#a94442",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "color": "#000000",
                  "fontSize": 17,
                  "height": 36,
                  "marginBottom": 5,
                  "paddingHorizontal": 7,
                  "paddingVertical": 7,
                },
                "normal": Object {
                  "borderColor": "#cccccc",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "color": "#000000",
                  "fontSize": 17,
                  "height": 36,
                  "marginBottom": 5,
                  "paddingHorizontal": 7,
                  "paddingVertical": 7,
                },
                "notEditable": Object {
                  "backgroundColor": "#eeeeee",
                  "borderColor": "#cccccc",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "color": "#777777",
                  "fontSize": 17,
                  "height": 36,
                  "marginBottom": 5,
                  "paddingHorizontal": 7,
                  "paddingVertical": 7,
                },
              },
              "textboxView": Object {
                "error": Object {},
                "normal": Object {},
                "notEditable": Object {},
              },
            }
          }
          templates={
            Object {
              "checkbox": [Function],
              "datepicker": [Function],
              "list": [Function],
              "select": [Function],
              "struct": [Function],
              "textbox": [Function],
            }
          }
          type={true}
          value={
            Object {
              "Passenger": "0",
              "Passenger1": "",
              "Passenger2": "",
              "Passenger3": "",
              "Passenger4": "",
              "Passenger5": "",
            }
          }
        />,
        "style": Object {
          "alignItems": "center",
          "flex": 1,
          "marginTop": 10,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "i18n": Object {
            "add": "Add",
            "down": "↓",
            "optional": " (optional)",
            "remove": "✘",
            "required": "",
            "up": "↑",
          },
          "onChange": [Function],
          "options": Object {
            "fields": Object {
              "Comment": Object {
                "label": "Kommentar (Valgfritt)",
                "multiline": true,
                "numberOfLines": 3,
              },
              "Passenger": Object {
                "label": "Antall passasjerer:",
                "nullOption": Object {
                  "text": "Velg antall Passasjerer",
                  "value": "",
                },
              },
              "Passenger1": Object {
                "label": "Passasjer 1:",
              },
              "Passenger2": Object {
                "label": "Passasjer 2:",
              },
              "Passenger3": Object {
                "label": "Passasjer 3:",
              },
              "Passenger4": Object {
                "label": "Passasjer 4:",
              },
              "Passenger5": Object {
                "label": "Passasjer 5:",
              },
            },
          },
          "stylesheet": Object {
            "button": Object {
              "alignSelf": "stretch",
              "backgroundColor": "#48BBEC",
              "borderColor": "#48BBEC",
              "borderRadius": 8,
              "borderWidth": 1,
              "height": 36,
              "justifyContent": "center",
              "marginBottom": 10,
            },
            "buttonText": Object {
              "alignSelf": "center",
              "color": "white",
              "fontSize": 18,
            },
            "checkbox": Object {
              "error": Object {
                "marginBottom": 4,
              },
              "normal": Object {
                "marginBottom": 4,
              },
            },
            "controlLabel": Object {
              "error": Object {
                "color": "#a94442",
                "fontSize": 17,
                "fontWeight": "500",
                "marginBottom": 7,
              },
              "normal": Object {
                "color": "#000000",
                "fontSize": 17,
                "fontWeight": "500",
                "marginBottom": 7,
              },
            },
            "dateTouchable": Object {
              "error": Object {},
              "normal": Object {},
            },
            "dateValue": Object {
              "error": Object {
                "color": "#a94442",
                "fontSize": 17,
                "marginBottom": 5,
                "padding": 7,
              },
              "normal": Object {
                "color": "#000000",
                "fontSize": 17,
                "marginBottom": 5,
                "padding": 7,
              },
            },
            "datepicker": Object {
              "error": Object {
                "marginBottom": 4,
              },
              "normal": Object {
                "marginBottom": 4,
              },
            },
            "errorBlock": Object {
              "color": "#a94442",
              "fontSize": 17,
              "marginBottom": 2,
            },
            "fieldset": Object {},
            "formGroup": Object {
              "error": Object {
                "marginBottom": 10,
              },
              "normal": Object {
                "marginBottom": 10,
              },
            },
            "helpBlock": Object {
              "error": Object {
                "color": "#999999",
                "fontSize": 17,
                "marginBottom": 2,
              },
              "normal": Object {
                "color": "#999999",
                "fontSize": 17,
                "marginBottom": 2,
              },
            },
            "pickerContainer": Object {
              "error": Object {
                "borderColor": "#a94442",
                "borderRadius": 4,
                "borderWidth": 1,
                "marginBottom": 4,
              },
              "normal": Object {
                "borderColor": "#cccccc",
                "borderRadius": 4,
                "borderWidth": 1,
                "marginBottom": 4,
              },
              "open": Object {},
            },
            "pickerTouchable": Object {
              "active": Object {
                "borderBottomWidth": 1,
                "borderColor": "#cccccc",
              },
              "error": Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 44,
              },
              "normal": Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 44,
              },
            },
            "pickerValue": Object {
              "error": Object {
                "fontSize": 17,
                "paddingLeft": 7,
              },
              "normal": Object {
                "fontSize": 17,
                "paddingLeft": 7,
              },
            },
            "select": Object {
              "error": Object {},
              "normal": Object {},
            },
            "textbox": Object {
              "error": Object {
                "borderColor": "#a94442",
                "borderRadius": 4,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 17,
                "height": 36,
                "marginBottom": 5,
                "paddingHorizontal": 7,
                "paddingVertical": 7,
              },
              "normal": Object {
                "borderColor": "#cccccc",
                "borderRadius": 4,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 17,
                "height": 36,
                "marginBottom": 5,
                "paddingHorizontal": 7,
                "paddingVertical": 7,
              },
              "notEditable": Object {
                "backgroundColor": "#eeeeee",
                "borderColor": "#cccccc",
                "borderRadius": 4,
                "borderWidth": 1,
                "color": "#777777",
                "fontSize": 17,
                "height": 36,
                "marginBottom": 5,
                "paddingHorizontal": 7,
                "paddingVertical": 7,
              },
            },
            "textboxView": Object {
              "error": Object {},
              "normal": Object {},
              "notEditable": Object {},
            },
          },
          "templates": Object {
            "checkbox": [Function],
            "datepicker": [Function],
            "list": [Function],
            "select": [Function],
            "struct": [Function],
            "textbox": [Function],
          },
          "type": true,
          "value": Object {
            "Passenger": "0",
            "Passenger1": "",
            "Passenger2": "",
            "Passenger3": "",
            "Passenger4": "",
            "Passenger5": "",
          },
        },
        "ref": [Function],
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
